runner/android_test_orchestrator/java/androidx/test/orchestrator/AndroidTestOrchestrator.java

```diff
@@ -401,7 +403,15 @@ private String getOutputFile() {
     if (null == test) {
       return TEST_COLLECTION_FILENAME;
     } else {
-      return String.format(TEST_RUN_FILENAME, test);
+      // Max filename length is 255 characters, so tests with longer names have to be truncated to prevent crashes.
+      // To prevent collisions, we append a timestamp to the filename.
+      // Every file also has the dot and "txt" appended, so effectively the max length is 255 - 4 = 251 characters.
+
+      String millis = Long.toString(System.currentTimeMillis());
+      String truncatedTest = test.substring(0, Math.min(test.length(), TEST_NAME_MAX_LENGTH - millis.length() - 1));
+      String name = String.format(TEST_RUN_FILENAME, truncatedTest + "_" + millis);
```

Comment:
probably should only append if the test name was longer than max?

Reply:
Probably yes - I just wanted to get it working asap :)

will do

Reply:
done